src/plugins/home/server/services/new_instance_status.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { IScopedClusterClient, SavedObjectsClientContract } from '../../../../core/server';
+import type { IndexPatternSavedObjectAttrs } from '../../../data/common/index_patterns/index_patterns';
+
+const LOGS_INDEX_PATTERN = 'metrics-*';
+const METRICS_INDEX_PATTERN = 'logs-*';
+
+const INDEX_PREFIXES_TO_IGNORE = [
+  'metrics-elastic_agent', // ignore index created by Fleet server itself
+  'logs-elastic_agent', // ignore index created by Fleet server itself
+];
+
+interface Deps {
+  esClient: IScopedClusterClient;
+  soClient: SavedObjectsClientContract;
+}
+
+export const isNewInstance = async ({ esClient, soClient }: Deps): Promise<boolean> => {
+  const indexPatterns = await soClient.find<IndexPatternSavedObjectAttrs>({
+    type: 'index-pattern',
+    fields: ['title'],
+    search: `*`,
+    searchFields: ['title'],
+    perPage: 1,
```

Comment:
`perPage` needs to be larger number to show all results for filtering through them on L42